interop/src/clients/corecrypto/ios.rs

```diff
@@ -0,0 +1,373 @@
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+
+use crate::{
+    clients::{EmulatedClient, EmulatedClientProtocol, EmulatedClientType, EmulatedMlsClient},
+    CIPHERSUITE_IN_USE,
+};
+use base64::{engine::general_purpose, Engine as _};
+use color_eyre::eyre::Result;
+use core_crypto::prelude::{KeyPackage, KeyPackageIn};
+use std::cell::{Cell, RefCell};
+use std::fmt::{Display, Formatter};
+use std::fs;
+use std::io::{BufRead, BufReader};
+use std::process::{Child, ChildStdout, Command, Stdio};
+use std::rc::Rc;
+use thiserror::Error;
+use tls_codec::Deserialize;
+
+#[derive(Debug)]
+struct SimulatorDriver {
+    device: String,
+    process: Child,
+    output: Rc<RefCell<BufReader<ChildStdout>>>,
+}
+
+#[derive(Debug, serde::Deserialize)]
+enum InteropResult {
+    #[serde(rename = "success")]
+    Success { value: String },
+    #[serde(rename = "failure")]
+    Failure { message: String },
+}
+
+#[derive(Error, Debug)]
+struct SimulatorDriverError {
+    msg: String,
+}
+
+impl Display for SimulatorDriverError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "simulator driver error: {}", self.msg)
+    }
+}
```

Comment:
You can use the `thiserror` crate to implement `Display`:

```suggestion
   #[error("simulator driver error: {msg}")]
    msg: String,
}
```